dspace-uclouvain/src/main/java/org/dspace/uclouvain/discovery/SolrServiceWorkflowMetadataRestrictionsPlugin.java

```diff
@@ -0,0 +1,94 @@
+package org.dspace.uclouvain.discovery;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.discovery.DiscoverQuery;
+import org.dspace.discovery.SearchServiceException;
+import org.dspace.discovery.SolrServiceSearchPlugin;
+import org.dspace.eperson.EPerson;
+import org.dspace.profile.ResearcherProfile;
+import org.dspace.profile.service.ResearcherProfileService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+/**
+ * Solr plugin used to add a or many filter(s) on the degree code of the workflow items.
+ * First, we extract them from the current eperson's metadata.
+ * After that we add them to the query filters.
+ * If a user has no degree code, nothing should be returned from Solr. 
+ */
+public class SolrServiceWorkflowMetadataRestrictionsPlugin implements SolrServiceSearchPlugin {
+
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    ResearcherProfileService researcherProfileService;
+
+    private String degreeMetadataFilterFieldName = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("uclouvain.solr.plugin.workflow.degree.field.filter", "masterthesis.degree.code_s");
```

Comment:
code_s --> code_sort. 
Do you check what happens if the solr key is unavailable ?